internal/hooks/HOOKS.md

```diff
@@ -0,0 +1,386 @@
+# Hooks Guide
+
```

Comment:
@meowgorithm @andreynering  this documentation is generated by crush we need to probably give this another look.